# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1251018755 -7200
#      Sun Aug 23 11:12:35 2009 +0200
# Branch release-3-2-x
# Node ID 310063a874079095b0e3ea456863980304f6e59f
# Parent  634697062fe15adbb8659fd850e5ffb7335ebc9f
zero matrix assignment fix

diff --git a/liboctave/Array-util.cc b/liboctave/Array-util.cc
--- a/liboctave/Array-util.cc
+++ b/liboctave/Array-util.cc
@@ -450,16 +450,22 @@ zero_dims_inquire (const idx_vector& i, 
 {
   bool icol = i.is_colon (), jcol = j.is_colon ();
   dim_vector rdv;
   if (icol && jcol && rhdv.length () == 2)
     {
       rdv(0) = rhdv(0);
       rdv(1) = rhdv(1);
     }
+  else if (rhdv.length () == 2
+           && ! i.is_scalar () && ! j.is_scalar ())
+    {
+      rdv(0) = icol ? rhdv(0) : i.extent (0);
+      rdv(1) = jcol ? rhdv(1) : j.extent (0);
+    }
   else
     {
       dim_vector rhdv0 = rhdv;
       rhdv0.chop_all_singletons ();
       int k = 0;
       rdv(0) = i.extent (0);
       if (icol)
         rdv(0) = rhdv0(k++);
diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,12 @@
+2009-08-19  Jaroslav Hajek  <highegg@gmail.com>
+
+	* Array-util.cc (zero_dims_inquire): Add matching ndims case.
+
 2009-08-16  Jaroslav Hajek  <highegg@gmail.com>
 
 	* dMatrix.cc, fMatrix.cc, CMatrix.cc, fCMatrix.cc: Add more tests.
 	* CMatrix.cc (xgemm): Fix vector * matrix case.
 	* fCMatrix.cc (xgemm): Ditto.
 
 2009-08-16  Jaroslav Hajek  <highegg@gmail.com>
 
